tests/compat/test_gecko51.py

```diff
@@ -0,0 +1,148 @@
+        expected = (
+            'The methods setAndFetchFaviconForPage and '
+            'replaceFaviconDataFromDataURL now default to using a null '
+            'principal for security reasons. An appropriate principal should '
+            'be passed if different behavior is required.')
+
+        script = '''
+            var ioService = Components.classes["@mozilla.org/network/io-service;1"].
+                  getService(Components.interfaces.nsIIOService);
+            var faviconUri = ioService.newURI('http://mozilla.org/fav.ico', 'utf-8', 'mozilla.org');
+
+            var faviconService = Components.classes["@mozilla.org/browser/favicon-service;1"].
+                     getService(Components.interfaces.nsIFaviconService).
+                     QueryInterface(Components.interfaces.mozIAsyncFavicons);
+            faviconService.replaceFaviconDataFromDataURL(faviconUri, 'data:base64/png');
+        '''
+
+        self.run_script_for_compat(script)
+
+        assert not self.compat_err.notices
+        assert not self.compat_err.errors
+
+        warning = [msg for msg in self.compat_err.warnings if msg['message'] == expected]
+        assert warning
+        assert warning[0]['compatibility_type'] == 'warning'
+
+    def test_BrowserOpenNewTabOrWindow_removed(self):
+        """https://github.com/mozilla/amo-validator/issues/473"""
+        expected = (
+            'The function BrowserOpenNewTabOrWindow has been removed.')
+
+        script = '''
+            window.BrowserOpenNewTabOrWindow();
+        '''
+        self.run_script_for_compat(script)
+
+        assert not self.compat_err.notices
+        assert not self.compat_err.errors
+
+        warning = [
+            msg for msg in self.compat_err.warnings
+            if msg['message'].startswith(expected)]
+        assert warning
+        assert warning[0]['compatibility_type'] == 'error'
+
+    def test_mozVisibilityState_unprefixed(self):
+        """https://github.com/mozilla/amo-validator/issues/474"""
+        expected = (
+            'The mozVisibilityState and mozHidden properties are no longer '
+            'prefixed.')
+
+        script = '''
+            var x = document.createElement('div');
+            div.mozVisibilityState = true;
+        '''
+        self.run_script_for_compat(script)
+
+        assert not self.compat_err.notices
+        assert not self.compat_err.errors
+
+        warning = [
+            msg for msg in self.compat_err.warnings
+            if msg['message'].startswith(expected)]
+        assert warning
+        assert warning[0]['compatibility_type'] == 'error'
+
+    def test_mozHidden_unprefixed(self):
+        """https://github.com/mozilla/amo-validator/issues/474"""
+        expected = (
+            'The mozVisibilityState and mozHidden properties are no longer '
+            'prefixed.')
+
+        script = '''
+            var x = document.createElement('div');
+            div.mozHidden = true;
```

Comment:
even though it probably doesn't make a difference for the test but `div` should be `x` in this example or am I reading this wrong?